### PR TITLE
add sysId example

### PR DESCRIPTION
**Pull request by LucienMorey**
Currently a WIP...

Outstanding tasks:

- [x] Finish Port of robot.py
- [x] confirm sysid log function is correct
- [x] convert docs strings
- [x] run on robot

### CODE REVIEW

**Comment by LucienMorey** [comment]
I don't have a tank drive handy to test this properly but I did adapt it for my team's swerve drive and deploy it there. feel free to request changes but I think this is good to go as a baseline

**Comment by auscompgeek** [comment]
Please add the robot to the test list.

File: SysId/subsystems/drive.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        )
+        # Record a frame for the right motors.  Since these share an encoder, we consider
+        # the entire group to be one motor.
+        sys_id_routine.motor("drive-right").voltage(
+            self.right_motor.get() * RobotController.getBatteryVoltage()
+        ).position(self.right_encoder.getDistance()).velocity(
+            self.right_encoder.getRate()
+        )
+
+    def arcadeDriveCommand(self, fwd: float, rot: float) -> Command:
```
Comment (auscompgeek) [inline SysId/subsystems/drive.py:85]: This should take lambdas, see https://github.com/wpilibsuite/allwpilib/blob/84ef71ace0648a71bd6e60e16c38be75914bb59d/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/sysid/subsystems/Drive.java#L118-L122

File: SysId/sysidroutinebot.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        Should be called during :meth:`.Robot.robotInit`.
+
+        Event binding methods are available on the :class:`.Trigger` class.
+        """
+
+        # Control the drive with split-stick arcade controls
+        self.drive.setDefaultCommand(
+            self.drive.arcadeDriveCommand(
+                lambda: self.controller.getLeftY(), lambda: self.controller.getRightX()
```
Comment (auscompgeek) [inline SysId/sysidroutinebot.py:42]: The signs should match the upstream example.
```suggestion
                lambda: -self.controller.getLeftY(), lambda: -self.controller.getRightX()
```

File: SysId/subsystems/drive.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    def __init__(self) -> None:
+        # The motors on the left side of the drive
+        self.left_motor = PWMSparkMax(DriveConstants.kLeftMotor1Port)
+        # TODO this has been disabled because the method is broken. see https://github.com/robotpy/mostrobotpy/issues/64
+        # self.left_motor.addFollower(PWMSparkMax(DriveConstants.kLeftMotor2Port))
+
+        # The motors on the right side of the drive
+        self.right_motor = PWMSparkMax(DriveConstants.kRightMotor1Port)
+        # TODO this has been disabled because the method is broken. see https://github.com/robotpy/mostrobotpy/issues/64
+        # self.right_motor.addFollower(PWMSparkMax(DriveConstants.kRightMotor2Port))
```
Comment (auscompgeek) [inline SysId/subsystems/drive.py:31]: `addFollower` is fixed.
```suggestion
        self.left_motor.addFollower(PWMSparkMax(DriveConstants.kLeftMotor2Port))

        # The motors on the right side of the drive
        self.right_motor = PWMSparkMax(DriveConstants.kRightMotor1Port)
        self.right_motor.addFollower(PWMSparkMax(DriveConstants.kRightMotor2Port))
```

**Review by auscompgeek — CHANGES_REQUESTED** [review]

**Review by auscompgeek — APPROVED** [review]
LGTM, thanks!

**Pull request merged**